### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by jhriggs**
I will be running `gonsul` in Docker. Have you considered creating an image and pushing it to Docker Hub? Here is a simple `Dockerfile` that builds using `golang:1-alpine` and then makes a gonsul image using `alpine`.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+ARG GONSUL=/go/src/github.com/miniclip/gonsul
+
+FROM golang:1-alpine as build
```
Comment (eduardev) [inline Dockerfile:3]: Let's pin version
```suggestion
FROM golang:1.14.3-alpine3.11 as build
```

**Comment by eduardev** [comment]
Never thought about this, but definitely great addition

**Comment by dpgaspar** [comment]
just merge it @eduardev  ;)

**Comment by eduardev** [comment]
@dpgaspar now that you left devops and became a developer you don't care about pinning versions? shame on you :P

**Comment by dpgaspar** [comment]
ahahaha, nop! still pinning everything

**Comment by fabiomsouto** [comment]
LGTM, thank you for the addition!

**Pull request merged**